src/VisualStudio/Core/Test.Next/Services/AssetStorageTests.cs

```diff
@@ -47,17 +43,30 @@ public void TestGetAssets()
             Assert.True(storage.TryGetAsset(checksum, out stored));
         }
 
-        private class MyAssetSource : AssetSource
+        [Fact, Trait(Traits.Feature, Traits.Features.RemoteHost)]
+        public async Task TestCleanup()
         {
-            public MyAssetSource(AssetStorage assetStorage, int sessionId) :
-                base(assetStorage, sessionId)
-            {
-            }
+            var storage = new AssetStorage(cleanupInterval: TimeSpan.FromMilliseconds(1), purgeAfter: TimeSpan.FromMilliseconds(2));
+
+            var checksum = new Checksum(Guid.NewGuid().ToByteArray());
+            var data = new object();
+
+            Assert.True(storage.TryAddAsset(checksum, data));
 
-            public override Task<IList<(Checksum, object)>> RequestAssetsAsync(int serviceId, ISet<Checksum> checksums, CancellationToken cancellationToken)
+            for (var i = 0; i < 10; i++)
             {
-                throw new NotImplementedException();
+                await Task.Delay(10);
```

Comment:
Delays in tests suck.  This code potentially delays for 100 seconds.  Why can't we use any of our standard mechanisms for listening for async things to happen in tests?  i.e. Waiters and whatnot?